internal/engine/eval/trusty/trusty.go

```diff
@@ -59,119 +60,203 @@ func NewTrustyEvaluator(ctx context.Context, ghcli provifv1.GitHub) (*Evaluator,
 		zerolog.Ctx(ctx).Info().Str("trusty-endpoint", trustyEndpoint).Msg("using trusty endpoint from environment")
 	}
 
+	piCli := newPiClient(trustyEndpoint)
+	if piCli == nil {
+		return nil, fmt.Errorf("failed to create pi client")
+	}
+
 	return &Evaluator{
 		cli:      ghcli,
 		endpoint: trustyEndpoint,
+		client:   piCli,
 	}, nil
 }
 
 // Eval implements the Evaluator interface.
-//
-//nolint:gocyclo
 func (e *Evaluator) Eval(ctx context.Context, pol map[string]any, res *engif.Result) error {
-	var lowScoringPackages []string
+	// Extract the dependency list from the PR
+	prDependencies, err := readPullRequestDependencies(res)
+	if err != nil {
+		return fmt.Errorf("reading pull request dependencies: %w", err)
+	}
+	if len(prDependencies.Deps) == 0 {
+		return nil
+	}
+
+	logger := zerolog.Ctx(ctx).With().
+		Int64("pull-number", prDependencies.Pr.Number).
+		Str("repo-owner", prDependencies.Pr.RepoOwner).
+		Str("repo-name", prDependencies.Pr.RepoName).Logger()
+
+	// Parse the profile data to get the policy configuration
+	ruleConfig, err := parseRuleConfig(pol)
+	if err != nil {
+		return fmt.Errorf("parsing policy configuration: %w", err)
+	}
+
+	prSummaryHandler, err := newSummaryPrHandler(prDependencies.Pr, e.cli, e.endpoint)
+	if err != nil {
+		return fmt.Errorf("failed to create summary handler: %w", err)
+	}
+
+	// Classify all dependencies, tracking all that are malicious or scored low
+	for _, dep := range prDependencies.Deps {
+		if err := classifyDependency(ctx, &logger, e.client, ruleConfig, prSummaryHandler, dep); err != nil {
+			return fmt.Errorf("classifying dependency: %w", err)
```

Comment:
Not blocking, but do you think we should error out the whole evaluation here? I wonder if we could classify a dependency as something like unknown instead.

Reply:
Since this logic is responsible for blocking PRs introducing malware into the codebase, I would rather build a more resilient client to make the requests more robust and indeed fail if we can't get the trusty score instead of just letting it through as an unknown. WDYT? 

The only two cases where this might fail are because of an error talking to trusty or a misconfiguration in the profile.